pkg/frontend/adminactions/clusterresourceusage.go

```diff
@@ -0,0 +1,129 @@
+package adminactions
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	restclient "k8s.io/client-go/rest"
+	metricsv1beta1 "k8s.io/metrics/pkg/client/clientset/versioned"
+)
+
+// ContainerMetrics is used to store metrics for containers
+type ContainerMetrics struct {
+	Name   string `json:"name"`
+	CPU    string `json:"cpu"`
+	Memory string `json:"memory"`
+}
+
+// PodMetrics represents the metrics for a pod
+type PodMetrics struct {
+	Namespace  string             `json:"namespace"`
+	PodName    string             `json:"podName"`
+	Containers []ContainerMetrics `json:"containers"`
+}
+
+// NodeMetrics represents the metrics for a node
+type NodeMetrics struct {
+	NodeName         string  `json:"nodeName"`
+	CPUUsage         string  `json:"cpuUsage"`
+	MemoryUsage      string  `json:"memoryUsage"`
+	CPUPercentage    float64 `json:"cpuPercentage"`
+	MemoryPercentage float64 `json:"memoryPercentage"`
+}
+
+// calculatePercentage calculates the percentage of resource usage.
+func calculatePercentage(usage string, total int64) float64 {
+	usageInt, _ := resource.ParseQuantity(usage)
+	usageInt64 := usageInt.Value()
+	return float64(usageInt64) / float64(total) * 100
+}
+
+// TopPods fetches the resource usage for all pods (or across namespaces) and calculates their usage.
+func (k *kubeActions) TopPods(ctx context.Context, restConfig *restclient.Config, allNamespaces bool) ([]PodMetrics, error) {
+	ns := ""
+	if !allNamespaces {
+		ns = "default"
+	}
+
+	client, err := metricsv1beta1.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	podMetrics, err := client.MetricsV1beta1().PodMetricses(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var result []PodMetrics
+	for _, item := range podMetrics.Items {
+		var containers []ContainerMetrics
+		for _, c := range item.Containers {
+			containers = append(containers, ContainerMetrics{
+				Name:   c.Name,
+				CPU:    c.Usage.Cpu().String(),
```

Comment:
Looks like Cpu() returns a pointer. Can you please double check if the return is guaranteed to be non-nil? Otherwise I suggest perform a check first. Same with Memory().